src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Utility class to parse user input.
+ */
+public class Parser {
+
+    private Scanner sc;
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Parser contains the Storage, Tasks, and Ui classes and acts as an
+     * intermediary for them to interact.
+     * 
+     * @param storage Storage class
+     * @param tasks   TaskList class
+     * @param ui      Ui Class
+     */
+    public Parser(Storage storage, TaskList tasks, Ui ui) {
+        this.sc = new Scanner(System.in);
+        this.storage = storage;
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+
+    /**
+     * Instantiates the scanner and parses inputs.
+     * 
+     */
+    public void open() {
+        while (sc.hasNextLine()) {
+            String input = sc.nextLine();
+            try {
+                identifyInput(input);
+            } catch (DukeTaskException e1) {
+                ui.showDukeTaskError();
+            } catch (IndexOutOfBoundsException e) {
+                ui.showDukeEmptyListError();
+            } catch (Exception e) {
+                ui.showDukeGeneralError();
+            }
+        }
+    }
+
+    public void close() {
+        sc.close();
+    }
+
+    private void identifyInput(String input) throws DukeGeneralException, DukeTaskException {
+        Task temp;
```

Comment:
Can the scope of temp be smaller?

Reply:
I'm not sure how to make the scope of temp smaller, since it's used by todo, deadline and event. Please advise thanks!

Reply:
It looks like you are initializing temp separately for todo, deadline and event so you could consider declaring it within those blocks.